Tasks/AppCenterDistributeV2/appcenterdistribute.ts

```diff
@@ -0,0 +1,495 @@
+    tl.debug(`---- url: ${beginUploadUrl}`);
+
+    let headers = {
+        "Content-Type": "application/json",
+        "X-API-Token": token,
+        "User-Agent": userAgent,
+        "internal-request-source": "VSTS"
+    };
+    request.post({ url: beginUploadUrl, headers: headers }, (err, res, body) => {
+        responseHandler(defer, err, res, body, () => {
+            let response = JSON.parse(body);
+            let uploadInfo: UploadInfo = {
+                upload_id: response['upload_id'],
+                upload_url: response['upload_url']
+            }
+
+            defer.resolve(uploadInfo);
+        });
+    });
+
+    return defer.promise;
+}
+
+function uploadRelease(uploadUrl: string, file: string, userAgent: string): Q.Promise<void> {
+    tl.debug("-- Uploading release...");
+    let defer = Q.defer<void>();
+    tl.debug(`---- url: ${uploadUrl}`);
+    let headers = {
+        "User-Agent": userAgent,
+        "internal-request-source": "VSTS"
+    };
+    let req = request.post({ url: uploadUrl, headers: headers }, (err, res, body) => {
+        responseHandler(defer, err, res, body, () => {
+            tl.debug('-- File uploaded.');
+            defer.resolve();
+        });
+    });
+
+    let form = req.form();
+    form.append('ipa', fs.createReadStream(file));
+
+    return defer.promise;
+}
+
+function commitRelease(apiServer: string, apiVersion: string, appSlug: string, upload_id: string, token: string, userAgent: string): Q.Promise<string> {
+    tl.debug("-- Finishing uploading release...");
+    let defer = Q.defer<string>();
+    let commitReleaseUrl: string = `${apiServer}/${apiVersion}/apps/${appSlug}/release_uploads/${upload_id}`;
+    tl.debug(`---- url: ${commitReleaseUrl}`);
+    let headers = {
+        "X-API-Token": token,
+        "User-Agent": userAgent,
+        "internal-request-source": "VSTS"
+    };
+
+    let commitBody = { "status": "committed" };
+
+    request.patch({ url: commitReleaseUrl, headers: headers, json: commitBody }, (err, res, body) => {
+        responseHandler(defer, err, res, body, () => {
+            if (body && body['release_url']) {
+                defer.resolve(body['release_url']);
+            } else {
+                defer.reject(tl.loc("FailedToUploadFile"));
+            }
+        });
+    })
+
+    return defer.promise;
+}
+
+function publishRelease(apiServer: string, releaseUrl: string, isMandatory: boolean, releaseNotes: string, destinationIds: string[], token: string, userAgent: string) {
+    tl.debug("-- Mark package available.");
+    let defer = Q.defer<void>();
+    let publishReleaseUrl: string = `${apiServer}/${releaseUrl}`;
+    tl.debug(`---- url: ${publishReleaseUrl}`);
+
+    let headers = {
+        "X-API-Token": token,
+        "User-Agent": userAgent,
+        "internal-request-source": "VSTS"
+    };
+    const destinations = [];
+    for (let destinationId of destinationIds) {
+            destinations.push({"id": destinationId});
+    }
```

Comment:
Can you use `map` on the `destinationIds` array?

Reply:
Thanks!